### PR TITLE
Use Fixed point computation instead of floating point.

### PR DESCRIPTION
**Pull request by ithinuel**
Relies on https://github.com/rp-rs/rp-hal/pull/440

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -200,7 +200,22 @@ where
...
+        let bit_freq = 32 * bus_freq;
+        let mut int = clock_freq / bit_freq;
+        let rem = clock_freq - (int * bit_freq);
+        let frac = (rem * 256) / bit_freq;
+        assert_ne!(
+            int, 0,
+            "The bus frequency times 32 must be lower or equal to the system clock."
+        );
+
+        if int == 65536 && frac == 0 {
```
Comment (9names) [inline src/lib.rs:212]: What are we handling in this case? (maybe it deserves a comment)

File: src/lib.rs [hunk 2]
```diff
@@ -200,7 +200,22 @@ where
...
 
         // Configure the PIO state machine.
-        let div = clock_freq.to_Hz() as f32 / ((32 * bus_freq).to_Hz() as f32);
+        let bit_freq = 32 * bus_freq;
+        let mut int = clock_freq / bit_freq;
+        let rem = clock_freq - (int * bit_freq);
+        let frac = (rem * 256) / bit_freq;
+        assert_ne!(
+            int, 0,
+            "The bus frequency times 32 must be lower or equal to the system clock."
```
[the remark this answers is not in the document]
Comment (ithinuel) [reply src/lib.rs]: I updated the way the boundaries are checked, I hope that notation will make it clearer.

File: src/lib.rs (hunk 1 above)
Comment (ithinuel) [reply src/lib.rs]: Added a short comment. It may look a bit odd. Should I point to the doc? 🤔

**Review by 9names — APPROVED** [review]
Looks good to me

**Pull request merged**